frame/bags-list/src/lib.rs

```diff
@@ -195,6 +210,25 @@ pub mod pallet {
 			let _ = Pallet::<T>::do_rebag(&dislocated, current_weight);
 			Ok(())
 		}
+
+		/// Move the caller's Id directly in front of `lighter`.
+		///
+		/// The dispatch origin for this call must be _Signed_ and can only be called by the Id of
+		/// the account going in front of `lighter`.
+		///
+		/// Only works if
+		/// - both nodes are within the same bag,
+		/// - and `origin` has a greater `VoteWeight` than `lighter`.
+		#[pallet::weight(T::WeightInfo::put_in_front_of())]
+		pub fn put_in_front_of(origin: OriginFor<T>, lighter: T::AccountId) -> DispatchResult {
+			let heavier = ensure_signed(origin)?;
+			List::<T>::put_in_front_of(
+				&lighter,
+				&heavier,
+				Box::new(T::VoteWeightProvider::vote_weight),
```

Comment:
no reason to pass this function here -- you have access to it everywhere.

Reply:
2add927

Reply:
My original idea was to make List as decoupled as possible from the rest of the pallet